examples/validium/contracts/ValidiumContract.sol

```diff
@@ -58,10 +50,9 @@ contract ValidiumContract is Ownable {
                     }
                 }
             }
-            isMember := eq(leaf, rootHash)
-
+            // checks if the calculated root matches the expected root
+            // then, check if index was zeroed while calculating proof, else an invalid index was provided
+            isMember := and(eq(leaf, rootHash), iszero(index))
```

Comment:
Does this work with odd number of leaves? will index be 0?

Reply:
It will work any number of leaves yes, we just validate that index is 0 at the end, otherwise it means an invalid index (i.e. index > width) was passed.